weave

```diff
@@ -1590,23 +1621,30 @@ launch_router() {
         fi
     fi
 
+    # Generate a hash of the password so we can tell if it has changed.
+    # Salt the hash with the peer name to make it even harder to reverse
+    [ -n "$WEAVE_PASSWORD" ] && PASSWORD_HASH="--password-hash=$(echo $PEERNAME$WEAVE_PASSWORD | sha256sum | cut -c1-32)"
```

Comment:
* Is the hash truncation with `cut` accidental? You need `-c1-64` to get the whole thing
* Agree with @rade that this value should not appear in logs
* Not convinced that `--password-hash` is a good name for the router arg, even though it is undocumented - `--ignored-value`? `--opaque-identity`?